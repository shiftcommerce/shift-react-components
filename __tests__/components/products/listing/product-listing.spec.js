// Libraries
import React from 'react'

// Component
import ProductListing from '../../../../src/components/products/listing/product-listing'
import ProductMenu from '../../../../src/components/products/listing/product-menu'
import ProductMenuOptions from '../../../../src/components/products/listing/product-menu-options'
import SearchFilters from '../../../../src/components/search/search-filters'

// Fixtures
import searchResultState from '../../../fixtures/search-results-state'

// Mock out the next/config
jest.mock('next/config', () => () => ({
  publicRuntimeConfig: {
    ALGOLIA_API_KEY: 'FAKE-API-KEY',
    ALGOLIA_APP_ID: 'FAKE-APP-ID',
    ALGOLIA_INDEX_NAME: 'FAKE-INDEX'
  }
}))

test('renders correctly', () => {
  // Arrange
  const searchState = {
    query: 'tray'
  }

  const onSearchStateChange = (nextSearchState) => (nextSearchState)

  // Act
  const wrapper = shallow(
    <ProductListing
      resultsState={searchResultState}
      onSearchStateChange={onSearchStateChange}
      searchState={searchState}
    />
  )

  const rootTree = wrapper.find('c-product-listing')

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(rootTree.find(ProductMenu))
  expect(rootTree.find(ProductMenuOptions))
  expect(rootTree.find(SearchFilters))
})
