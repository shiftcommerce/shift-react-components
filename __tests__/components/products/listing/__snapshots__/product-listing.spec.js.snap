// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Fragment>
  <AlgoliaStateResults(ProductMenu) />
  <Breadcrumb />
  <div
    className="c-product-listing-wrapper"
  >
    <SearchFilters
      filtersShown={false}
      toggleFiltering={[Function]}
    />
    <div
      className="c-product-listing"
    >
      <div
        className="c-product-listing__menu"
      >
        <ProductMenuOptions
          toggleFiltering={[Function]}
        />
      </div>
      <AlgoliaInfiniteHits(SearchResults) />
    </div>
  </div>
</Fragment>
`;
